bazel/thirdparty/seastar.BUILD

```diff
@@ -576,9 +576,19 @@ cc_library(
     }) + select({
         ":with_debug": [
             "SEASTAR_DEBUG",
+            "SEASTAR_DEBUG_PROMISE",
             "SEASTAR_DEBUG_SHARED_PTR",
+            "SEASTAR_TYPE_ERASE_MORE",
         ],
         "//conditions:default": [],
+    }) + select({
+        # This isn't the best way to check this, but is the only way I can
+        # currently think of to replicate this behavior in Bazel. In CMake
+        # seastar only enables this if CMAKE_BUILD_SHARED_LIBS is enabled.
+        "@//bazel:optimized_build": [],
+        "//conditions:default": [
+            "SEASTAR_BUILD_SHARED_LIBS",
```

Comment:
oh... yeh it is a shame that this macro is used in the code. surely they didn't really need to do that, hmmm

Reply:
Yeah agreed, it's used in quite a few places tho

https://github.com/search?q=repo%3Ascylladb%2Fseastar%20SEASTAR_BUILD_SHARED_LIBS&type=code

Reply:
Most of it seems to be around optimizing removing initialization locks

Reply:
In case you need to fill some idle time https://github.com/scylladb/seastar/pull/2629#discussion_r1933971633 ...